lib/VCMIDirs.cpp

```diff
@@ -530,7 +527,63 @@ bfs::path VCMIDirsOSX::libraryPath() const { return "."; }
 bfs::path VCMIDirsOSX::binaryPath() const { return "."; }
 #endif // VCMI_IOS, VCMI_MAC
 #elif defined(VCMI_XDG)
-class VCMIDirsXDG : public IVCMIDirsUNIX
+class VCMIDirsELF : public IVCMIDirsUNIX
+{
+public:
+	std::string libraryName(const std::string & basename) const override { return "lib" + basename + ".so"; }
+};
+
+// android is treated differently to XDG
```

Comment:
can you come up with a better name by chance? Didn't want to use "Linux" because BSDs also use XDG.

Reply:
Personally I would rather get rid of "VCMIDirsELF" class and copy libraryName method as it into both VCMIDirsAndroid and VCMIDirsXDG 

Yes, this would duplicate 1 line of 1 method, but IMO more clear - this file is already hard to understand due to countless ifdef's and inheritance chains

Reply:
ok, will do. In the future we can split this file into `VCMIDirs_Windows.cpp` etc.